python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,921 @@
+
+### Data Types
+
+Strings are an example of a **data type** in Python. Knowing what data type you're working with is important, because certain functions and methods will only work on certain data types, or will behave differently depending on the data type. The built-in function `type()` will return as output the data type of its argument.
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print(type("Hello World!"))
+
+</script>
+</lia-keep>
+</div>
+
+When we pass the phrase "Hello World!" to the `type()` function, we get the output `<class 'str'>`, "str" being short for "string". A **string** is a sequence of characters; you tell Python that a sequence is a string by surrounding it in quotation marks. You can use either single or double quotes, but it's a good idea to be consistent. Strings are a specific data type in Python; other data types include integers, floats (or decimal numbers), lists, and boolean (like True or False, which may look like strings but are a special data type and don't require quotation marks), but there are many more. Try editing the code cell above, replacing "Hello World" with different inputs (for example "Hi", 9.8, 10, or False) and run the cell again to see how the output changes.
+
+<div class = "important">
+Notice that in the above example, we have passed the `type("Hello World!")` function call as an argument to `print()`. Nesting functions like this is common (you can do the same with methods), but there is another option we'll discuss in the next section-- using variables!
+</div>
+
+### Quiz: Functions and methods
+
+Which of the following are valid examples of Python code? Select all that apply. Use the code cell below to check your knowledge by replacing the `x` argument in the `print()` function with code you're curious about.
+
+[[X]] `len("Python is awesome!")`
+[[X]] `"python".upper()`
+[[X]] `type("4.3")`
+[[X]] `type(True)`
+***
+<div class = "answer">
+These are all examples of valid Python code! `print(type(4))` returns the type of the argument `4` (which is `int`, short for integer) and prints that output to the screen. `"python.upper()"` makes the string "python" all uppercase. `type("4.3")` might be a little tricky, since 4.3 is a decimal number (or **float**) but it's in quotation marks-- but those quotation marks just mean that, in this case, "4.3" is a string! `type(True)` might also be tricky, since at first it looks like a string that doesn't have the required quotation marks. However, `True` and `False` are special in Python, and have the type `boolean`.  
+</div>
+***
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print(x)
+
+</script>
+</lia-keep>
+</div>
+
+## Variables
+
+Rather than passing a value (like a string or a number) directly to a function, we can also assign values to variables and pass those to functions. **Variables** are ways to store values and objects for use later.
```

Comment:
Emphasize the word "assign" in this line for learners who aren't used to this usage.